lib/webauth.js

```diff
@@ -157,7 +163,40 @@ module.exports = function(authManager) {
         result.addHandlerResult(status, handler);
       }
       res.status(200).json(result);
-    },
+    }),
+
+    refreshStatus: Promise.coroutine(function*(req, res) {
```

Comment:
You did the _authenticateOrRefresh() thing in zss-auth although the authenticate and refresh logic is quite different. Here you just copied doLogin(). It and refreshStatus() differ in just a couple lines, so I would say it makes more sense to make a generic function as well.